src/main/java/org/broadinstitute/hellbender/tools/sv/LocusDepthtoBAF.java

```diff
@@ -0,0 +1,270 @@
+     * the depth of the alt call as a fraction of the total depth is returned.
+     */
+    @VisibleForTesting double calcBAF( final LocusDepth locusDepth,
+                                       final int refIndex, final int altIndex ) {
+        final int totalDepth = locusDepth.getTotalDepth();
+        if ( totalDepth < minTotalDepth ) {
+            return BafEvidence.MISSING_VALUE;
+        }
+        double expectRefAlt = totalDepth / 2.;
+        final double altDepth = locusDepth.getDepth(altIndex);
+        final double refDiff = locusDepth.getDepth(refIndex) - expectRefAlt;
+        final double altDiff = altDepth - expectRefAlt;
+        double chiSq = (refDiff * refDiff + altDiff * altDiff) / expectRefAlt;
+        double fitProb = 1. - chiSqDist.cumulativeProbability(chiSq);
+        if ( fitProb < minHetProbability ) {
+            return BafEvidence.MISSING_VALUE;
+        }
+        return altDepth / totalDepth;
+    }
+
+    private boolean sameLocus( final Locatable locus ) {
+        if ( sameLocusBuffer.isEmpty() ) {
+            return true;
+        }
+        final LocusDepth curLocus = sameLocusBuffer.get(0);
+        return curLocus.getContig().equals(locus.getContig()) &&
+                curLocus.getStart() == locus.getStart();
+    }
+
+    private void processBuffer() {
+        final VariantContext vc = nextSite();
+        if ( !sameLocus(vc) ) {
+            final Locatable loc = sameLocusBuffer.get(0);
+            throw new UserException("expecting locus " + loc.getContig() + ":" + loc.getStart() +
+                    ", but found locus " + vc.getContig() + ":" + vc.getStart() + " in baf sites vcf");
+        }
+        final List<Allele> alleles = vc.getAlleles();
+        final int refIndex = Nucleotide.decode(alleles.get(0).getBases()[0]).ordinal();
+        if ( refIndex > 3 ) {
+            throw new UserException("ref call is not [ACGT] in vcf at " + vc.getContig() + ":" + vc.getStart());
+        }
+        final int nAlleles = alleles.size();
+        final int[] altIndices = new int[nAlleles - 1];
+        for ( int alleleIndex = 1; alleleIndex < nAlleles; ++alleleIndex ) {
+            final int altIndex = Nucleotide.decode(alleles.get(1).getBases()[0]).ordinal();
+            if ( altIndex > 3 ) {
+                throw new UserException("alt call is not [ACGT] in vcf at " + vc.getContig() + ":" + vc.getStart());
+            }
+            altIndices[alleleIndex - 1] = altIndex;
+        }
+        final List<BafEvidence> beList = new ArrayList<>(sameLocusBuffer.size());
+        for ( final LocusDepth ld : sameLocusBuffer ) {
+            for ( final int altIndex : altIndices ) {
+                double baf = calcBAF(ld, refIndex, altIndex);
+                if ( baf != BafEvidence.MISSING_VALUE ) {
+                    beList.add(new BafEvidence(ld.getSample(), ld.getContig(), ld.getStart(), baf));
+                    break;
+                }
+            }
```

Comment:
In some cases this will lead to bias. I didn't think through some of the problems multi-allelics are going to cause us with compatibility here. Apologies for flip-flopping on this, but let's stick to bi-allelic for both LD collection and BAF generation. Regardless, it seems like this implementation would be problematic in some edge cases. I think we will need to either provide the sites list to this tool, or go back to storing the ref/alt alleles in LD (which would be fine, probably simpler and faster).

Reply:
I'll remove the multi-allelic option, which kind of weirded me out anyway.
However, I'm confused by the last part of your comment:  we *do* provide the sites list to this tool.  That's where I'm getting the ref/alt call info, since it's no longer in the LD records.  (See lines 210-223.)